tests/common/osutil/test_default.py

```diff
@@ -88,14 +86,22 @@ def test_mount_dvd_failure(self, _):
                         self.assertTrue(msg in ustr(ose))
                         self.assertTrue(patch_run.call_count == 6)
 
-    def test_get_first_if(self):
+
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil.get_primary_interface', return_value='eth0')
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil._get_all_interfaces', return_value={'eth0':'10.0.0.1'})
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil.is_loopback', return_value=False)
```

Comment:
There would be more utility to this mock if it actually looked at the interface name (which is an argument to the method) and returned True if and only if the name started with "lo". You would supply the same mock to test_get_first_if_nosuchprimary(); it would ensure that, no matter what Python does to its dictionary construction, the mocked interface dictionary `{'eth0':'192.0.2.100','lo':'127.0.0.1'}` would never pick 'lo' if the primary was not found.

Reply:
Outside the test class:
```
def fake_is_loopback(iface):
    return iface.startswith('lo')
```

Then the patch becomes
```
@patch('azurelinuxagent.common.osutil.default.DefaultOSUtil.is_loopback', fake_is_loopback) 
```

Supply that patch to both test_get_first_if() and test_get_first_if_nosuchprimary().